_posts/2024-08-25-mot-tracking-metrics.md

```diff
@@ -0,0 +1,172 @@
+
+The computation of MOTA involves a temporal dependency. It penalizes cases where track assignments change between consecutive frames. An identity switch is counted if a ground truth target i is matched to track j in the current frame but was matched to a different track k (where k ≠ j) in the previous frame.
+
+ In [TrackEval code](https://github.com/JonathonLuiten/TrackEval/blob/master/trackeval/metrics/clear.py#L81){:target="_blank"}{:rel="noopener noreferrer"} this is done using a simple gating trick:
+
+```yaml
+score = IoU(GT, pred)
+if pred != previous_assigned_id(GT):
+    score = score * 1000
+```
+
+The MOTA metric is computed across all frames as:
+
+$$
+\text{MOTA} = 1 - \frac{\sum_t (FN_t + FP_t + IDSW_t)}{\sum_t GT_t}
+$$
+
+Where $t$ is the frame index, FN are False Negatives, FP are False Positives, IDSW are Identity Switches, and GT is the number of ground truth objects.
+
+While MOTA's simplicity is appealing, it has some limitations:
+
+1. It only considers the **previous frame** for IDSW, so each switch is penalized **only once**, regardless of how long the incorrect assignment persists.
+2. It can be **dominated by FP and FN** in crowded scenes, making IDSW less impactful.
+3. **IoU is fixed** so more or less detection accuracy is not reflected on the metric
+
+## IDF1
+
+IDF1 addresses some of MOTA's limitations by focusing on how long the tracker correctly identifies an object, rather than just counting errors. It's based on the concept of Identification Precision (IDP) and Identification Recall (IDR).
+
+It computes the assignment between prediction and ground truth objects across the entire video, rather than frame by frame. 
+
+The metric is simple:
+
+$$
+\text{IDF1} = \frac{2 * \text{IDTP}}{2 * \text{IDTP} + \text{IDFP} + \text{IDFN}}
+$$
+
+Where:
+
+- **IDTP** (ID True Positive): The number of correctly identified detections
+- **IDFP** (ID False Positive): Tracker hypotheses that don't match any ground truth
+- **IDFN** (ID False Negative): Ground truth trajectories that aren't tracked
+
+The global assignment is computed using the Hungarian algorithm. It pick the best combination between prediction and ground truth for the whole video. It is easier to understand by observing the image introduced in HOTA paper:
+
+
+<div class="post-center-image" style="max-width: 800px; margin: 0 auto;">
+    {% picture pimage /assets/images/fullsize/posts/2024-08-25-mot-tracking-metrics/idf1.png --alt IDF1 metric diagram %}
+</div>
+
+{:refdef: class="image-caption"}
+*Figure 3. IDF1 metric diagram*
+{: refdef}
+
+The main problem I see with IDF1 is finding the best one-to-one matching between predicted and ground truth trajectories for the entire sequence. However, this approach can sometimes oversimplify complex tracking scenarios:
```

Comment:
I think this sentence in itself doesn't make sense. "The main problem I see with IDF1 is finding the best one-to-one matching between predicted and ground truth trajectories for the entire sequence."

You mean that finding that best matching is a problem because it is computationally expensive? Or that it is sensitive to some initialization in the algorithm? Or that it does not necessarily reflect what you care about?

If it's the latter, I'd state it explicitly. "The main issue I see with IDF1 is that finding the best one-to-one matching between predicted and ground-truth trajectories may not be a good proxy for what a good tracker should predict. For instance, this approach can ..."